libflux/core/src/formatter/mod.rs

```diff
@@ -750,10 +750,36 @@ impl Formatter {
     }
 
     fn format_testcase_statement(&mut self, n: &ast::TestCaseStmt) {
+        let sep = '\n';
         self.format_comments(&n.base.comments);
         self.write_string("testcase ");
         self.format_node(&Node::Identifier(&n.id));
-        self.format_node(&Node::Block(&n.block));
+        self.write_rune('{');
+
+        if !n.imports.is_empty() || !n.body.is_empty() {
+            self.write_rune(sep);
+            self.indent();
+
+            for imp in n.imports.iter() {
+                self.write_indent();
+                self.format_import_declaration(imp);
+                self.write_rune(sep);
+            }
+
+            if !n.imports.is_empty() && !n.body.is_empty() {
+                self.write_rune(sep);
+            }
+
+            for stmt in n.body.iter() {
+                self.write_indent();
+                self.format_node(&Node::from_stmt(stmt));
+                self.write_rune(sep);
+            }
+
+            self.unindent();
+            self.write_indent();
+        }
+        self.write_rune('}');
```

Comment:
Can you add a test case for the formatting code here?

Reply:
Done.